scriptworker/github.py

```diff
@@ -119,13 +117,12 @@ def get_tag_hash(self, tag_name):
         url = '/'.join([repo.rstrip('/'), 'branch_commits', revision])
         from scriptworker.task import get_decision_task_id
         cache_key = '{}-{}'.format(get_decision_task_id(context.task), url)
-        async with _branch_commits_cache_lock:
```

Comment:
I'd love to get @eoger and @JohanLorenzo 's take on this change, since they put it in originally.

Reply:
Running the test in `scriptworker/test/test_production.py`  I was getting consistent errors (both locally and in CI) without removing this.

Reply:
Okay! Would you have CI logs to share, @tomprince? I've thought about it a bit more and I believe I remember what happened: without the lock, 4 requests were fired at once which was unnecessary.

Reply:
Okay! Would you have CI logs to share, @tomprince? I've thought about it a bit more and I believe I remember what happened: without the lock, 4 requests were fired at once which was unnecessary.

Reply:
I can't find the logs in travis, but if you checkout f71081e805227aa3a66b346d5cd027e6fff20014 and run `py.test -k 'test_verify_production_cot[branch_context10]'` you should see the error:
```
scriptworker/task.py:462: in is_try_or_pull_request
    return await is_pull_request(context, task)
scriptworker/task.py:443: in is_pull_request
    conditions.append(not await github_repository.has_commit_landed_on_repository(context, revision))
scriptworker/github.py:122: in has_commit_landed_on_repository
    async with _branch_commits_cache_lock:
/usr/lib64/python3.6/asyncio/locks.py:79: in __aenter__
    yield from self.acquire()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <asyncio.locks.Lock object at 0x7fd6a7bde128 [unlocked]>

    @coroutine
    def acquire(self):
        """Acquire a lock.
    
        This method blocks until the lock is unlocked, then sets it to
        locked and returns True.
        """
        if not self._locked and all(w.cancelled() for w in self._waiters):
            self._locked = True
            return True
    
        fut = self._loop.create_future()
        self._waiters.append(fut)
    
        # Finally block should be called before the CancelledError
        # handling as we don't want CancelledError to call
        # _wake_up_first() and attempt to wake up itself.
        try:
            try:
>               yield from fut
E               RuntimeError: Task <Task pending coro=<is_try_or_pull_request() running at /Depot/Mozilla/scriptworker/scriptworker/task.py:462> cb=[_wait.<locals>._on_completion() at /usr/lib64/python3.6/asyncio/tasks.py:380]> got Future <Future pending> attached to a different loop
```